### PR TITLE
Create User#questions method, closes #935

### LINKED ISSUE #935: Create a @user.questions method to return a user's questions
Hi, this is a `first-timers-only issue`. This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

If you have contributed before, consider leaving this one for someone new, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

## The problem

We need to fetch and display the [questions](https://publiclab.org/questions) for a given user (like [on their profile](https://publiclab.org/profile/warren)!) in a few different places, but don't have an easy way to do it -- we write a chained ActiveRecord method like here:

https://github.com/publiclab/plots2/blob/master/app/controllers/users_controller.rb#L119

It looks like:

```ruby
DrupalNode.questions.where(status: 1, uid: @user.id)
```

## Solution

We should make a method in the User model to make this easier, and reduce repetition. The user model is in: https://github.com/publiclab/plots2/tree/master/app/models/user.rb

We can add a new method (at the bottom of the file is fine) like:

```ruby
def questions
  DrupalNode.questions.where(status: 1, uid: @user.id)
end
```

Then we could replace these two lines: https://github.com/publiclab/plots2/blob/master/app/controllers/users_controller.rb#L119-L120

with this (the double "user" is unfortunate, but has to do with some legacy issues in our database setup):

```ruby
  @questions = @user.user.questions
```

We should also write a unit test for this new function, in: https://github.com/publiclab/plots2/blob/master/test/unit/user_test.rb

Right after line 26, we could add:

```ruby
    assert user.questions.count > 0
```

This would confirm that, in our tests, our new method correctly returns that the user "jeff" has more than 0 questions.

## Steps to Fix

- [ ] claim this issue with a comment here, below, and ask any clarifying questions you need
- [ ] set up a repository locally following the README instructions, and make sure that all tests pass
- [ ] try to fix the issue following the steps above, but even before you're done, you can:
- [ ] commit your changes and start a pull request (see [contributing to Public Lab software](https://publiclab.org/wiki/contributing-to-public-lab-software)) but mark it as "in progress" if you have questions or if you haven't finished
- [ ] alert someone via the developers list (see below) to have your pull request merged. This may take a few extra steps depending on when you do it, but we'll help you out!

Please email the developers list (see https://publiclab.org/developers) if you have questions!

### PR DESCRIPTION
**Pull request by ruan-brandao**
Refered to issue #935 

I still have some doubts about this feature. I left some coments regarding them.

@jywarren, @ebarry please take a look at this PR :smiley: 

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [ ] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: test/unit/user_test.rb [hunk 1]
```diff
@@ -26,6 +26,11 @@ class UserTest < ActiveSupport::TestCase
     assert_equal user.notes, user.drupal_user.notes
   end
 
+  test "user questions" do
+    user = rusers(:jeff)
+    assert user.questions.count.count > 0
```
Comment (ruan-brandao) [inline test/unit/user_test.rb]: I've found a problem here. the `questions` method is returning a `DrupalNode::FriendlyIdActiveRecordRelation`, so when i call `.count` it returns an `ActiveSupport::OrderedHash`, and it gives an error when I try to compare it with the number `0`.

How should solve this problem?

**Comment by jywarren** [comment]
Regarding the spaces, that's not a problem for me. As to the other question, that's odd -- `.count` is standard ActiveRecord, I thought. But perhaps since it's a joined query, it's returning multiple counts, and/or we can't just count a joined query? For simplicity we can simply use `.length` -- which is for array-like things (and doesn't generate a COUNT SQL query). 

I tested this out:

```
> DrupalNode.questions.length
  DrupalNode Load (0.7ms)  SELECT "node".* FROM "node" INNER JOIN "community_tags" ON "community_tags"."nid" = "node"."nid" INNER JOIN "term_data" ON "term_data"."tid" = "community_tags"."tid" WHERE "node"."type" = 'note' AND (term_data.name LIKE 'question:%') GROUP BY node.nid
 => 2 
> DrupalNode.questions.count
   (0.8ms)  SELECT COUNT(*) AS count_all, node.nid AS node_nid FROM "node" INNER JOIN "community_tags" ON "community_tags"."nid" = "node"."nid" INNER JOIN "term_data" ON "term_data"."tid" = "community_tags"."tid" WHERE "node"."type" = 'note' AND (term_data.name LIKE 'question:%') GROUP BY node.nid
 => {44=>1, 45=>1}
```

Sound good?

**Comment by jywarren** [comment]
Hi! It looks like the tests failed... Can you look into why? Check out the tests here: https://travis-ci.org/publiclab/plots2/builds/172818875

I'm sure we'll figure it out and I look forward to merging it, thanks!!

**Comment by ruan-brandao** [comment]
Okay @jywarren, aparently there is an error in this test `test_should_get_profile(UsersControllerTest)`
I'm trying to reproduce the error locally to get more info abut it.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -116,8 +116,7 @@ def profile
                        .page(params[:page])
                        .order("nid DESC")
                        .where(status: 1, uid: @user.uid)
-    @questions = DrupalNode.questions
-                           .where(status: 1, uid: @user.uid)
+    @questions = @user.user.questions
```
Comment (jywarren) [inline app/controllers/users_controller.rb]: I'm so sorry, I believe I was mistaken that a double `user` was needed. This is a User record, not a DrupalUser record, so we should be fine just calling simply `@user.questions`. I believe that'll fix the test!

**Comment by jywarren** [comment]
OK, my bad -- i left an inline comment with the fix, and if you add one more commit to this PR (just add a commit and push it up to the same branch as the first one) the tests will re-run and you should be OK to go!

Thanks for your patience, looks like this one was a tad more complex than we expected; sometimes that happens I guess!

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (ruan-brandao) [inline app/controllers/users_controller.rb]: I think that's not it. I tried that but when I ran `rake test:all`a lot of errors like that popped out:

```NoMethodError: undefined method `questions' for #<DrupalUsers:0x00555a52b4a828>```

**Comment by jywarren** [comment]
Hmm, this is an odd one. I'll see if I can reproduce the error on my machine, hang on a sec.

I noticed also, though, that you submitted a pull request from your master branch. That can be an issue if `publiclab/plots2`'s master branch goes out of sync with your master branch, because you can't rebase your changes on top or pull in the latest changes (with nowhere left to put them, since you've added to your own master branch). It's not a huge deal, but there's a simple way around this, which is doing any new work of your own in a named feature branch. 

There's [a guide to this workflow here](https://publiclab.org/wiki/contributing-to-public-lab-software#A+sample+git+workflow) and if you need to rewind your master branch to get it back in sync with publiclab's, you can [follow this procedure](https://publiclab.org/wiki/contributing-to-public-lab-software#Rewinding+the+master+branch) -- I'm happy to help you with either, and can also log into your cloud9 environment to help if you run into trouble (you'd have to invite me).

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (jywarren) [inline app/controllers/users_controller.rb]: You're right -- this line (https://github.com/publiclab/plots2/pull/945/files#diff-4e05ad0d64e6100656b63ad1e78f32c5R112) shows that we're actually using a DrupalUsers record. Hmm. 

Oh, i get it! I just added a `legacy_user` fixture -- that has no corresponding User record, to test some other code. The test is:

https://github.com/publiclab/plots2/blob/master/test/functional/users_controller_test.rb#L81

```ruby
    get :profile, id: DrupalUsers.where(status: 1).last.name
```

So it's getting that legacy user fixture. Change `last` to `first` and it should start to work!

**Comment by jywarren** [comment]
OK, you're rebased, and tests are running now, and it's merge-able. I think you just have to add that last change to the test:

https://github.com/publiclab/plots2/blob/master/test/functional/users_controller_test.rb#L81

```ruby
    get :profile, id: DrupalUsers.where(status: 1).last.name
```

Change `.last` to `.first` and it should start to work! Fingers Xed!

**Comment by ruan-brandao** [comment]
:tada: The build passed :tada:

**Comment by jywarren** [comment]
Perfect -- merging, and thanks!!

**Comment by jywarren** [comment]
And if you're looking for a new challenge, we'd love to have your involvement on anything in this list: https://github.com/publiclab/plots2/labels/help-wanted

Also, apologies if I asked already, but how did you hear about us? We're looking to improve our outreach to new contributors. Thanks again!

**Comment by ruan-brandao** [comment]
Thanks. I'll see if I can help with another issue on this list.

I've heard about you guys through [this tweet](https://twitter.com/PublicLab/status/793163027302518788) that was retweeted by [@yourfirstpr](https://twitter.com/yourfirstpr).

**Pull request merged**